sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeysEventSource.cs

```diff
@@ -40,7 +40,12 @@ public void KeyTypeNotSupported(string operation, KeyVaultKey key)
         {
             if (IsEnabled())
             {
-                string keyType = key?.KeyType.ToString() ?? "(null)";
+                string keyType = "(null)";
+                if (key != null)
+                {
+                    keyType = key.KeyType.ToString();
+                }
```

Comment:
```suggestion
                string keyType = key?.KeyType.ToString() ?? "(null)"';
```

Reply:
I changed the code this way because `key.KeyType` itself threw a `NullReferenceException`. It references, effectively, what is contained on `key`. The code path shouldn't actually be hit with other changes I made, but this was a defense-in-depth change to make sure our tracing itself doesn't throw.